File: FE/salt/src/service/CartService.js
import axios from "axios"

export const getCartDetail = async(username) => {
    const res = await axios.get(`http://localhost:8080/cart/cart-detail?username=${username}`)
    console.log("res servuce", res);
    return res;
}
export const addCart = async (quantity, username, productId) => {

 const res = await axios.post(`http://localhost:8080/cart/add-cart?quantity=${quantity}&username=${username}&productId=${productId}`)
 return res;
}
export const deleteCart = async(username, id) => {
    await axios.delete(`http://localhost:8080/cart/delete-cart?username=${username}&id=${id}`)
}
export const increase = async(username, id) => {
   const res= await axios.post(`http://localhost:8080/cart/increase-quantity?username=${username}&id=${id}`)
   console.log("in", res);
}
export const decrease = async(username, id) => {
    await axios.post(`http://localhost:8080/cart/decrease-quantity?username=${username}&id=${id}`)
}
export const getAllDelivery = async () => {
    const res = await axios.get("http://localhost:8080/cart/delivery")
    return res.data
}

File: BE/spring2/src/main/java/com/example/spring2/cart/service/CartService.java
package com.example.spring2.cart.service;

import com.example.spring2.app_user.model.AppUser;
import com.example.spring2.app_user.repository.IAppUserRepository;
import com.example.spring2.cart.model.Cart;
import com.example.spring2.cart.model.ICartDto;
import com.example.spring2.cart.repository.ICartRepository;
import com.example.spring2.delivery.model.Delivery;
import com.example.spring2.delivery.repository.IDeliveryRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CartService implements ICartService{
    @Autowired
    private ICartRepository cartRepository;
    @Autowired
    IAppUserRepository appUserRepository;
    @Autowired
    private IDeliveryRepository deliveryRepository;
    @Override
    public List<ICartDto> getCartDetail(String username) {
        AppUser appUser = appUserRepository.getAccountByUserName(username);
        return cartRepository.getCartDetail(appUser.getId());
    }

    @Override
    public void addCart(Integer quantity, String username, Long productId) {
        AppUser appUser = appUserRepository.getAccountByUserName(username);
        cartRepository.AddCart(quantity,appUser.getId(),productId);
    }

    @Override
    public void deleteCart(Long id, String username) {
        AppUser appUser = appUserRepository.getAccountByUserName(username);
        cartRepository.deleteCart(id, appUser.getId());
    }

    @Override
    public Long getCartById(String username, Long productId) {
        AppUser appUser = appUserRepository.getAccountByUserName(username);
        return cartRepository.getCartByIds(appUser.getId(),productId);
    }

    @Override
    public void increaseQuantity(String username, Long id, Integer quantity) {
        AppUser appUser = appUserRepository.getAccountByUserName(username);
        cartRepository.increaseQuantity(appUser.getId(), id, quantity);
    }

    @Override
    public void decreaseQuantity(String username, Long id) {
        AppUser appUser = appUserRepository.getAccountByUserName(username);
        cartRepository.decreaseQuantity(appUser.getId(), id);
    }

    @Override
    public List<Delivery> getAllDelivery() {
        return deliveryRepository.getAllDelivery();
    }


}
